openpdf/src/main/java/com/lowagie/text/pdf/PdfBoolean.java

```diff
@@ -50,87 +50,43 @@
- * <P>
+ * Wrapper around {@code boolean} value.
+ *
  * This object is described in the 'Portable Document Format Reference Manual version 1.7'
  * section 3.2.1 (page 52).
  *
- * @see        PdfObject
- * @see        BadPdfFormatException
+ * @see PdfBoolean#TRUE
+ * @see PdfBoolean#FALSE
+ * @see PdfBoolean#fromValue(boolean)
  */
-
-public class PdfBoolean extends PdfObject {
-    
-    // static membervariables (possible values of a boolean object)
-    public static final PdfBoolean PDFTRUE = new PdfBoolean(true);
-    public static final PdfBoolean PDFFALSE = new PdfBoolean(false);
-/** A possible value of <CODE>PdfBoolean</CODE> */
-    public static final String TRUE = "true";
-    
-/** A possible value of <CODE>PdfBoolean</CODE> */
-    public static final String FALSE = "false";
-    
-    // membervariables
-    
-/** the boolean value of this object */
-    private boolean value;
-    
-    // constructors
+public final class PdfBoolean extends PdfObject {
     
-/**
- * Constructs a <CODE>PdfBoolean</CODE>-object.
- *
- * @param        value            the value of the new <CODE>PdfObject</CODE>
- */
+    public static final PdfBoolean TRUE = new PdfBoolean(true);
+    public static final PdfBoolean FALSE = new PdfBoolean(false);
+
+    private static final String TRUE_STR = "true";
+    private static final String FALSE_STR = "false";
+
+    public static PdfBoolean fromValue(final boolean value) {
+        return new PdfBoolean(value);
+    }
+
+    private final boolean value;
     
-    public PdfBoolean(boolean value) {
+    private PdfBoolean(final boolean value) {
         super(BOOLEAN);
-        if (value) {
-            setContent(TRUE);
-        }
-        else {
-            setContent(FALSE);
-        }
+        setContent(value ? TRUE_STR : FALSE_STR);
         this.value = value;
     }
-    
-/**
- * Constructs a <CODE>PdfBoolean</CODE>-object.
- *
- * @param        value            the value of the new <CODE>PdfObject</CODE>, represented as a <CODE>String</CODE>
- *
- * @throws        BadPdfFormatException    thrown if the <VAR>value</VAR> isn't '<CODE>true</CODE>' or '<CODE>false</CODE>'
- */
-    
-    public PdfBoolean(String value) throws BadPdfFormatException {
-        super(BOOLEAN, value);
-        if (value.equals(TRUE)) {
-            this.value = true;
-        }
-        else if (value.equals(FALSE)) {
-            this.value = false;
-        }
-        else {
-            throw new BadPdfFormatException(MessageLocalization.getComposedMessage("the.value.has.to.be.true.of.false.instead.of.1", value));
-        }
-    }
-    
-    // methods returning the value of this object
-    
-/**
- * Returns the primitive value of the <CODE>PdfBoolean</CODE>-object.
- *
- * @return        the actual value of the object.
- */
-    
-    public boolean booleanValue() {
+
+
+    boolean booleanValue() {
```

Comment:
This should remain public, as this can break user code.

Reply:
Ok, I will deprecate mentioned changes for removal, because this class must not be a part of public API in any way and even be accessible by the user code